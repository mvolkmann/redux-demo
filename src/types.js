// @flow

export type ActionType = {
  type: string,
  payload?: mixed
};

export type DispatchType = (action: ActionType) => void;

export type StateType = {
  counter: number,
  delta: number
};

export type SubstateType = {
  counter?: number,
  delta?: number
};
