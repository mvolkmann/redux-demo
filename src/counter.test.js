// @flow

import React from 'react';
import Counter from './counter';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import './types';

/**
 * These tests just verify that when the user
 * interacts with the UI in a certain way,
 * the expected Redux actions are dispatched.
 */
describe('Counter', () => {
  let store;

  beforeEach(() => {
    const mockStore = configureStore();
    const initialState = {counter: 0, delta: 1};
    store = mockStore(initialState);
  });

  test('should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Counter />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test.skip('should decrement', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Counter />
      </Provider>
    );
    const btn = wrapper.find('.dec-btn');
    btn.simulate('click');

    const actions = store.getActions();
    expect(actions[0]).toEqual({type: 'decrement'});
  });

  test('should increment', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Counter />
      </Provider>
    );
    const btn = wrapper.find('.inc-btn');
    btn.simulate('click');

    const actions = store.getActions();
    expect(actions[0]).toEqual({type: 'increment'});
  });
});
