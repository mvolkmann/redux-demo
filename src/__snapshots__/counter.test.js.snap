// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div
  className="counter"
>
  <div
    className="button-row"
  >
    <button
      className="dec-btn"
      onClick={[Function]}
    >
      -
    </button>
    0
    <button
      className="inc-btn"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="delta"
  >
    <label>
      Delta
    </label>
    <input
      onChange={[Function]}
      type="number"
      value={1}
    />
  </div>
</div>
`;
